src/economy/portdock.cc

```diff
@@ -502,6 +382,32 @@ uint32_t PortDock::count_waiting(const PortDock* dest) const {
 	return waiting_.size();
 }
 
+uint32_t PortDock::calc_max_priority(const EditorGameBase& egbase, const PortDock& dest) const {
+	uint32_t p = 0;
+	for (const ShippingItem& si : waiting_) {
+		if (si.destination_dock_.serial() == dest.serial()) {
+			WareInstance* ware = nullptr;
+			Worker* worker = nullptr;
+			si.get(egbase, &ware, &worker);
+			++p;
+			if (ware) {
+				assert(!worker);
+				if (ware->get_transfer() && ware->get_transfer()->get_request()) {
+					// I don't know when this shouldn't be true,
```

Comment:
Whats the semantic of this condidtion?

Reply:
This only checks that we really have an associated Transfer which we can query for a priority. In the rare edge-case that a ware/worker has no Transfer (does not normally happen during a game) we just assume the lowest priority of 0.